File: c-udemy/diagonal-negativos.c
#include <stdio.h>

int main() {

    int ordemMatriz;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &ordemMatriz);

    int matriz[ordemMatriz][ordemMatriz], count = 0;

    for (int i = 0; i < ordemMatriz; i++) {
        for (int j = 0; j < ordemMatriz; j++) {
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &matriz[i][j]);
        }
    }

    printf("Diagonal principal: \n");
    for (int i = 0; i < ordemMatriz; i++) {
        for (int j = 0; j < ordemMatriz; j++) {
            if (i == j) {
                printf("%d ", matriz[i][j]);
            }
        }
    }

    for (int i = 0; i < ordemMatriz; i++) {
        for (int j = 0; j < ordemMatriz; j++) {
            if (matriz[i][j] < 0) {
                count += 1;;
            }
        }
    }

    printf("\nQuantidade de negativos = %d\n", count);

    return 0;
}

File: python-udemy/diagonal-negativos.py
ordemMatriz = int(input("Qual a ordem da matriz? "))

matriz = [[0 for x in range(ordemMatriz)] for x in range(ordemMatriz)]

for i in range(0, ordemMatriz):
    for j in range(0, ordemMatriz):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("Diagonal principal:")
for i in range(0, ordemMatriz):
    for j in range(0, ordemMatriz):
        if i == j:
            print(f"{matriz[i][j]} ", end="")

count =0

print()
for i in range(0, ordemMatriz):
    for j in range(0, ordemMatriz):
        if matriz[i][j] < 0:
            count += 1

print(f"Quantida de negativos = {count}")